tests/Unit/ExternalVersionUpdate/UpdateTest.php

```diff
@@ -0,0 +1,183 @@
+<?php
+/**
+ * Unit tests related to external version update.
+ */
+
+namespace WooCommerce\Facebook\Tests\ExternalVersionUpdate;
+
+use WooCommerce\Facebook\Framework\Plugin\Exception as PluginException;
+use WooCommerce\Facebook\Framework\Api\Exception as ApiException;
+use \WP_UnitTestCase;
+
+/**
+ * The External version update unit test class.
+ */
+class UpdateTest extends WP_UnitTestCase {
+
+	/**
+	 * Instance of the Update class that we are testing.
+	 *
+	 * @var \WooCommerce\Facebook\ExternalVersionUpdate\Update The object to be tested.
+	 */
+	private $update;
+
+	/**
+	 * Setup the test object for each test.
+	 */
+	public function setUp():void {
+		$this->update = new \WooCommerce\Facebook\ExternalVersionUpdate\Update();
+	}
+
+	/**
+	 * Test should_update_version
+	 */
+	public function test_should_update_version() {
+		$plugin = facebook_for_woocommerce();
+
+		// Assert update not required when the versions match.
+		update_option( 'facebook_for_woocommerce_latest_version_sent_to_server', \WC_Facebookcommerce_Utils::PLUGIN_VERSION );
+		$should_update = $this->update->should_update_version();
+		$this->assertFalse( $should_update );
+
+		/**
+		 * Set the $plugin->connection_handler and $plugin->api access to true. This will allow us
+		 * to assign the mock objects to these properties.
+		 */
+		$plugin_ref_obj          = new \ReflectionObject( $plugin );
+		$prop_connection_handler = $plugin_ref_obj->getProperty( 'connection_handler' );
+		$prop_connection_handler->setAccessible( true );
+
+		// Connection Handler mock object to return is_connected as false.
+		$mock_connection_handler = $this->getMockBuilder( \WooCommerce\Facebook\Handlers\Connection::class )
```

Comment:
Minor: I won't comment on everything else, but I think every reference outside of this namespace should be imported.